in_toto/runlib.go

```diff
@@ -76,12 +91,25 @@ func RecordArtifacts(paths []string) (map[string]interface{}, error) {
 				if info.IsDir() {
 					return nil
 				}
-				artifact, err := RecordArtifact(path)
-				// Abort if artifact can't be recorded, e.g. due to file permissions
-				if err != nil {
-					return err
+				if exclude_patterns != nil {
```

Comment:
Could you please encapsulate this inside a function to improve grokability? I'm also wondering if we can do something to avoid this arrowhead pattern. Probably having the function behave like a filter. I also suspect you can restructure this to remove some of the repeated code.

Reply:
Ok, I'll work on these changes. Also, it seems that updating the Go version on AppVeyor wasn't enough to resolve the build fail issue. It's still failing with the same reason on AppVeyor. Though, the build is passing on Travis with Go 1.12.